### DISCUSSION TITLE
Excited to see active development!

### DISCUSSION THREAD

**Discussion by tbartholomaus** [scatseisnet/scatseisnet#17] [General]
Hi Scatseisnet team,

A team here at the University of Idaho has been working with learning the methods from your 2020 paper over the last month or two, including first trying to get going with scatseis, and have just installed scatseisnet via pip (https://pypi.org/project/scatseisnet/).  We've been poking around your githubs and just noticed that it looks like you're working (even as I type!) on making the package even more accessible and under formal development for broader use. This is super exciting.  On our team is Grace Barcheck, who I understand has communicated with you before, Yoram Terleth, and Verenis Lucas (both Ph.D. students I'm working with).

Anyway, we're poking around again, and are starting through your new tutorials (so far so good), but we'd love to help test anything you're working on, or help you in any way if we can.

Thanks for your work on this, and sharing it with the research community.

------------------------------------------------
Timothy C. Bartholomaus ([he/him](https://www.mypronouns.org/what-and-why))

Associate Professor
Dept. of Earth and Spatial Sciences - University of Idaho
[1-208-885-6217](tel:1%C2%A0208%20885-6217) - [tbartholomaus@uidaho.edu](mailto:tbartholomaus@uidaho.edu) - [www.tbartholomaus.org](http://www.tbartholomaus.org/)

**Comment by Lchuang** [comment]
Hello also to the Scatseisnet team,

Here is a quick shoutout and echo Dr. Bartholomaus's excitement about the new developments on the package. As an old user (although I've been lurking in the shadows), I am super excited to see the latest developments!

Thanks for your hard work and dedication to improving the package. 
I'm stoked to see what the latest updates offer and would love to contribute or give feedback if I can.

Cheers,
Lindsay Chuang 
Geophysics Ph.D. candidate
@EAS, Georgia Tech

**Comment by ReneSteinmann** [comment]
Hello,

I just had a call with Leonard and he mentioned that there are these lovely and motivating comments on GitHub, so nice to see that, thanks a lot! It definitely keeps up the motivation :) 

We hope to release soon a new package with pypi (hopefully in April). The old package on pypi works with the command line interface and we decided to remove it, since it masks a lot of inner workings of scatseisnet. So the new package will be much slimmer and the notebooks are there to show how scatseisnet can be used. If any of you have feedback or ideas, you are very much welcome to share them.

Best,
René Steinmann

**Comment by tbartholomaus** [comment]
Very cool- yeah, we were talking yesterday about how nice it is to have the notebooks to explore, and interact with the output along the way (relative to the CLI, which we'd just gotten started with).  I'm afraid we lost a little time with [Leonard's scatnet repo](https://github.com/leonard-seydoux/scatnet) until we realized there were these newer versions, but it's been great to learn, including with the more recent papers too.  Anyway, we'll be following!  Thanks.

**Comment by Lchuang** [comment]
The dependency issues were one of the biggest obstacles I encountered while using the old package. Particularly some functions the package called in tensorflow 1.x were either moved or deprecated for a later version of tensorflow. It will be beneficial if the new version of the package gives clearer guidance on the versions of libraries used. Perhaps some tests or documentation could go in that direction (either provided by the developers or reported by users). :)

**Comment by leonard-seydoux** [comment]
Greetings everyone,

Thank you so much for your support, it makes our work much more enjoyable! It's great to (e-)meet all of you!

As Timothy mentioned, the repository he referred to is currently outdated due to the deprecation of many packages from TensorFlow 1. I am sorry that you lost time on it, I just indicated there that the repository is outdated, which I should have done a little while ago! Unfortunately, this includes some of the experimental packages that were developed therein, such as the differentiable Gaussian Mixture Model, which we lost in TensorFlow 2. Currently, we do not have the necessary funding to bring this version to TensorFlow 2 (and maintain it through the future major releases), but if could be nice to do so.

We will put our efforts on this new version and striving to make it as clean and documented as possible, including testing, and to maintain it as far as we can. We also welcome your contributions and are grateful to Timothy for resolving the issue with the reduced_type parameter since the pull request. Your branch is now outdated due to our modifications, Timothy, but we will still manage a clean merge. We will do our best to respond more promptly from now on. 😄

We hope to have the opportunity to meet all of you in person one day!

All the best,
Leonard
